crates/binstalk/src/fetchers.rs

```diff
@@ -107,9 +127,100 @@ impl Data {
     }
 }
 
+impl RepoInfo {
+    /// If `repo` contains a subcrate, then extracts and returns it.
+    /// It will also remove that subcrate path from `repo` to match
+    /// `scheme:/{repo_owner}/{repo_name}`
+    fn detect_subcrate(repo: &mut Url, repository_host: RepositoryHost) -> Option<String> {
+        match repository_host {
+            RepositoryHost::GitHub => Self::detect_subcrate_common(repo, &["tree", "main"]),
```

Comment:
wait, that might not always be `main`. eg could be `master`, `trunk`, `develop`, and even more arbitrary branch names

Reply:
Oh right, I'm fooled by myself as I look at these names literally instead of them as branch name.
Then I suppose it will also applies to GitLab.

Reply:
Thanks, I've fixed this.